src/main/java/org/altbeacon/beacon/service/scanner/CycledLeScanner.java

```diff
@@ -40,7 +42,10 @@
     private long mScanPeriod;
 
     protected long mBetweenScanPeriod;
-    protected final Handler mHandler = new Handler();
+
+    protected final Handler mHandler = new Handler(Looper.getMainLooper());
+    protected final Handler mScanHandler;
```

Comment:
OK, got it now... apologies, it's been awhile, it's been awhile since I looked at this code.  I was incorrectly thinking that new `Handler()`  always creates a new thread.